src/components/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useRef, useEffect } from 'react';
+import { Button } from './Button';
+import { Todo } from '../types/Todo';
+import { USER_ID } from '../api/todos';
+import { addTodos } from '../api/todos';
+import { ErroMessage } from '../utils/errorMessages';
+
+type Props = {
+  todos: Todo[];
+  addTodo: (newTodo: Todo) => void;
+  handleErrorMessages: (newErrorMessage: ErroMessage) => void;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
```

Comment:
Please, remove `React.Dispatch` from your all property types, unless you actually use it

```suggestion
  setTempTodo: (value: Todo | null) => void;
```